File: src/main/java/kingdoms/server/entities/EntityBarracksKeeper.java
package kingdoms.server.entities;

import kingdoms.api.entities.EntityNPC;
import kingdoms.server.TaleOfKingdoms;
import net.minecraft.entity.player.EntityPlayer;
import net.minecraft.init.Items;
import net.minecraft.item.ItemStack;
import net.minecraft.util.ChatComponentTranslation;
import net.minecraft.world.World;

import static kingdoms.server.handlers.GuiHandler.GUI_WARDEN;

public final class EntityBarracksKeeper extends EntityNPC
{
    public EntityBarracksKeeper(World world)
    {
        super(world, new ItemStack(Items.iron_sword), 20.0F);
    }

    @Override
    public boolean canBePushed()
    {
        return false;
    }

    @Override
    public boolean interact(EntityPlayer player)
    {
        if (this.canInteractWith(player))
        {
            if (!this.worldObj.isRemote)
            {
                player.addChatMessage(new ChatComponentTranslation("npc.warden.dialog"));
            }
            player.openGui(TaleOfKingdoms.instance, GUI_WARDEN, worldObj, 0, 0, 0);
        }
        return true;
    }
}

File: src/main/java/kingdoms/server/entities/EntityBarracksKeeper.kt
package kingdoms.server.entities

import kingdoms.api.entities.EntityNPC
import kingdoms.server.handlers.NetworkHandler
import net.minecraft.entity.player.EntityPlayer
import net.minecraft.init.Items
import net.minecraft.item.ItemStack
import net.minecraft.util.ChatComponentTranslation
import net.minecraft.world.World

class EntityBarracksKeeper(world: World): EntityNPC(world, ItemStack(Items.iron_sword), 20.0F)
{
    override fun canBePushed(): Boolean
    {
        return false
    }

    override fun interact(player: EntityPlayer): Boolean
    {
        if (!worldObj.isRemote)
        {
            if (this.canInteractWith(player))
            {
                player.addChatMessage(ChatComponentTranslation("npc.warden.dialog"))
                NetworkHandler.openGui(this.entityId, player)
            }
        }
        return true
    }
}